src/dsp_tools/utils/connection_live.py

```diff
@@ -162,30 +236,34 @@ def post(
         # otherwise the client can get a timeout error while the API is still processing the request
         # in that case, the client's retry will have undesired side effects (e.g. duplicated resources),
         # and the response of the original API call will be lost
-        timeout = 60
+        timeout = timeout or 60
         if not route.startswith("/"):
             route = f"/{route}"
         url = self.server + route
         headers = {}
         if jsondata:
-            headers["Content-Type"] = f"{content_type}; charset=UTF-8"
+            headers["Content-Type"] = "application/json; charset=UTF-8"
         if self.token:
             headers["Authorization"] = f"Bearer {self.token}"
 
-        response = requests.post(
-            url=url,
-            headers=headers,
+        request = partial(requests.post, url=url, headers=headers, timeout=timeout)
```

Comment:
good idea, but I think I have an even better one: collect these keywords in a dict, and pass that dict as `try_network_action(requests.post, **kwargs)`.

And then, the signature of `try_network_action` could be simplified: 
 - accept only `Literal["GET", "POST", "PUT", "DELETE"]` for the parameter `action`. 
 - remove parameter `*args` 
 - turn parameter `**kwargs` into `kwargs` (mandatory)

Then, the logic of `try_network_action` can be much simplified.

Reply:
good idea. I was also having some thoughts on how to simplify `try_network_action`, which were going a slightly different direction. How about we leave this as is for now (I think it works like that) and discuss our ideas for improvements, which we can then implement as follow-up tasks?  
My aim (which I managed only partially) was to keep this PR as "mechanical" as possible, simply inverting the hierarchy of those two concepts, and not tackle any bigger improvements just yet. (Agile... baby steps... short-lived PRs... yada-yada 😄 )

Reply:
Sure, that's fine :-)